eng/templates/stages/release.yml

```diff
@@ -182,14 +182,25 @@ stages:
 
           echo -e "\n\nCreating GitHub release..."
 
-          gh release create "$(releaseTag)"         \
-            "$release_json#Release manifest"        \
-            --repo dotnet/dotnet                    \
-            --title "${{ parameters.releaseName }}" \
-            --target "$(dotnetDotnetCommit)"        \
-            --notes-file "$release_notes"           \
-            $prerelease                             \
-            $draft
+          set +e
+          result=$(gh release create "$(releaseTag)" \
+            "$release_json#Release manifest"         \
+            --repo dotnet/dotnet                     \
+            --title "${{ parameters.releaseName }}"  \
+            --target "$(dotnetDotnetCommit)"         \
+            --notes-file "$release_notes"            \
+            $prerelease                              \
+            $draft \
+            2>&1)
+
+          if [ $? -ne 0 ]; then
+            if [[ "$result" =~ "already exists" ]]; then
+              echo "##vso[task.logissue type=warning]GitHub release for $(dotnetDotnetCommit) already exists"
```

Comment:
Since this error is specific to a release with a duplicate tag name, I think the warning message should include the tag name.

Reply:
```suggestion
              echo "##vso[task.logissue type=warning]GitHub release tag $(releaseTag) already exists (https://github.com/dotnet/dotnet/tree/$(releaseTag))"
```